behaviours/meme_rater.py

```diff
@@ -62,74 +70,115 @@ def number_emoji(number: int):
-        elif embed.video and embed.video.proxy_url:
-            image_url = f"{embed.video.proxy_url}?format=webp&width={embed.video.width}&height={embed.video.height}"
-        if not image_url:
-            continue
-
-        try:
-            username = None
-            if event.author:
-                username = event.author.username
-            res = await get_meme_rating(image_url, username)
-            if not res:
-                continue
-            ratings.append(res.rate)
-            explanations.append(res.explanation)
-        except Exception:
-            continue
-    await rate_meme(event.message, ratings, explanations)
 
+possible_emojis: list[hikari.UnicodeEmoji | hikari.CustomEmoji] = [
+    number_emoji(i) for i in range(11)
+]
+possible_emojis.append(hikari.Emoji.parse("🐱"))
+possible_emojis.append(hikari.Emoji.parse("👍"))
+possible_emojis.append(hikari.Emoji.parse("💩"))
+possible_emojis.append(hikari.Emoji.parse("❓"))
 
-async def msg_create(event: hikari.GuildMessageCreateEvent) -> None:
-    if event.channel_id != MEME_CHANNEL_ID:
-        return
-    ratings = list[int]()
-    explanations = list[str]()
-
-    for attachment in event.message.attachments:
-        att_ext = (attachment.extension or "").lower()
-        logging.info(f"Attachment extension: {att_ext}")
-        if att_ext not in IMG_FILE_EXTENSIONS:
-            continue
-        image_url = attachment.url
-        try:
-            res = await get_meme_rating(image_url, event.author.username)
-            if not res:
-                continue
-            ratings.append(res.rate)
-            explanations.append(res.explanation)
-        except Exception:
-            continue
-
-    await rate_meme(event.message, ratings, explanations)
+
+def get_meme_stats(
+    message_id: hikari.Snowflake,
+) -> MemeStat | None:
+    cursor = db.cursor()
+    stats = cursor.execute(
+        "SELECT * FROM meme_stats WHERE message_id = ?",
+        (message_id,),
+    ).fetchone()
+    if not stats:
+        return None
+    db_meme_stats = MemeStat(
+        author_id=stats[0],
+        message_id=stats[1],
+        meme_score=stats[2],
+        timestamp=stats[3],
+        meme_rating=stats[4],
+        rating_count=stats[5],
+        meme_reasoning=stats[6],
+        emoji="",
+    )
+    return db_meme_stats
 
 
 async def rate_meme(
-    message: hikari.PartialMessage, ratings: list[int], explanations: list[str]
-) -> None:
+    message: hikari.PartialMessage, rating_results: list[agents.MemeAnswer]
+) -> MemeStat | None:
+    message = await message.app.rest.fetch_message(message.channel_id, message.id)
+    cursor = db.cursor()
+
+    is_already_rated = any(
+        reaction.emoji in possible_emojis for reaction in message.reactions
+    )
+    if is_already_rated:
+        return get_meme_stats(message.id)
+
     async with RATER_LOCK:
-        if not ratings or not explanations:
+        if (not rating_results) or (len(rating_results) == 0):
```

Comment:
Just `if not rating_results:` is sufficient here

Reply:
Done